HCCGo/app/js/jobViewCtrl.js

```diff
@@ -66,13 +92,92 @@ jobViewModule.controller('jobViewCtrl', ['$scope', '$log', '$timeout', 'connecti
                     $log.debug("Something went wrong updating the db.");
                   }
                 }
-              );
-            }
-          });
+                );
+              }
+            });
+          }
         });
       }
       $scope.job = result;
     });
   });
 
+  /**
+   * Saves the output or error to a file
+   * @method saveFile
+   * @memberof HCCGo.jobViewCtrl
+   * @param {String} fileType - Denotes either the output or error to be Saved
+   * @param {Event} $event - Used to stop the save event from moving up the DOM
+   */
+  $scope.saveFile = function(fileType, $event) {
+    const {dialog} = require('electron').remote;
+    const fs = require('fs');
+    const path = require('path');
+
+    var options = {
+      filters: [
+        { name: 'text', extensions: ['txt']}
+      ]
+    };
+
+    dialog.showSaveDialog(options, function(localFile) {
+      // If the user clicks cancel, localFile will be undefined
+      if (!localFile) return;
+
+      downloadRemoteFile($scope.job[fileType], localFile).then(function(flag) {
+        if (flag) {
+          fs.writeFile(localFile, $scope.job[fileType], function(err) {});
+        }
+        notifierService.success(path.basename(localFile).trim() + " was Successfully Saved!", "Save Successful!");
+      },
+      function(err) {
+        notifierService.error(msg);
+      });
+      
+    });
+    // Stop the propagation of the click
+    $event.stopPropagation();
+  };
+
+  /**
+   * Used to download the file requested from the server, since only 5MB worth of data is displayed to the user.
+   * @method downloadRemoteFile
+   * @memberof HCCGo.jobViewCtrl
+   * @param {String} remotePath - Denotes the path on the server to download the file from
+   * @param {String} localPath - Denotes the path where the user wants the file saved
+   * @returns {Promise} A promise denoting whether the file was downloaded from the server or not.
+   */
+  function downloadRemoteFile(remotePath, localPath) {
+    var deferred = $q.defer();
+
+    connectionService.getFileSize(remotePath).then(function(size) {
+      if (size > 5*1025*1024) {
+        connectionService.quickDownload(remotePath, localPath).then(function(flag) {
+          deferred.resolve(true);
+        }, 
+        function(err) {
+            deferred.reject("Error downloading file!");  
+        });
+      }
+      else {
+        deferred.resolve(false);
+      }
+      $log.log(deferred);
+      return deferred.promise;
+    });
+  }
```

Comment:
Need to return the defferred above this line.  Right now, you are returning it after checking the file size, and returning it to nowhere.